docs/probing.md

```diff
@@ -0,0 +1,318 @@
+
+Similar to the ```ProbabilisticScoringDecayParameters```, ```ProbabilisticScoringFeeParameters``` parameters are customizable, but defaults are provided. You can read more about the ```ProbabilisticScoringFeeParameters``` [here](https://docs.rs/lightning/latest/lightning/routing/scoring/struct.ProbabilisticScoringFeeParameters.html).
+
+## The ```ProbabilisticScorer```
+
+As discussed earlier, the ```ProbabilisticScorer``` (seen below) is responsible for maintaining and updating our node's estimates of available channel liquidity.
+
+A ```ProbabilisticScorer``` has the following fields:
+- ```decay_params```: This field holds the decay parameters (```ProbabilisticScoringDecayParameters```) for the scorer. As mentioned above, these are customizable, but LDK also provides defaults. You can learn more about them [here](https://docs.rs/lightning/latest/lightning/routing/scoring/struct.ProbabilisticScoringDecayParameters.html).
+- ```network_graph```: This field holds a reference to the network graph.
+- ```logger```: This field holds a reference to the logger.
+- ```channel_liquidities```: This field holds a ```HashMap```, which maps **short channel IDs** to ```ChannelLiquidity``` objects. These objects include liquidity estimates, historical data for the respective channels, and timestamps indicating when the liquidity estimates were last updated.
+
+Once implemented, the ```ProbabilisticScorer```, along with its parameters (```ProbabilisticScoringDecayParameters``` and ```ProbabilisticScoringFeeParameters```), becomes the backbone of path-finding in LDK. The next section will describe best practices for implementing, training, and serving the ```ProbabilisticScorer``` within your application.
+
+``` rust
+pub struct ProbabilisticScorer<G: Deref<Target = NetworkGraph<L>>, L: Deref>
+where L::Target: Logger {
+	decay_params: ProbabilisticScoringDecayParameters,
+	network_graph: G,
+	logger: L,
+	channel_liquidities: HashMap<u64, ChannelLiquidity>,
+}
+```
+
+## Best Practices
+
+Before digging into best practices for probing, it's important to note that popular Lightning Network implementations such as Core Lightning (CLN), Lightning Network Daemon (LND), and Lightning Development Kit (LDK), have different internal representations of the Lightning Network. Therefore, it's not possible to simply "feed" CLN, LND, or LDK the same scoring file or information when setting up path-finding functionality.
+
+### General Architecture Approaches
```

Comment:
@jkczyz does this check out? So probing isn't something we would do on a mobile phone?

Reply:
Yeah, the wallet's vendor would periodically provide an updated scorer for the wallet to use. A mobile wallet would need to be active to send / handle probes. However, a wallet could send a JIT probe before making a payment to help ensure it will make it to the destination. And it could update it's local scorer with recent payment success / failure results, though that information would tend to decay before the next payment, if payments are infrequent.